os/kernel/fault_manager/fault_manager.c

```diff
@@ -0,0 +1,285 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <tinyara/init.h>
+#include <tinyara/board.h>
+#include <debug.h>
+#include <fcntl.h>
+#include "fault_manager.h"
+#ifdef CONFIG_BOARDCTL_RESET
+#include <sys/boardctl.h>
+#include <stdlib.h>
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Data
+ ****************************************************************************/
+
+/* This structure is used to store the mapping between app id and application path
+ * Its value is filled at a time of registering the application for fault manager
+ */
+struct apptable reg_table[CONFIG_NUM_APPS] = { { -1, " "}, { -1, " "} };
```

Comment:
where is CONFIG_NUM_APPS?

Reply:
Added here @jeongarmy 
https://github.com/Samsung/TizenRT/pull/2747

Reply:
You should not use CONFIG value directly, please apply below style to prevent build error after change configs in menuconfig
```
#if (CONFIG_NUM_APPS < 2) 
#define CONFIG_NUM_APPS 2
#endif
struct apptable reg_table[CONFIG_NUM_APPS] = { { -1, " "}, { -1, " "} };
```

Reply:
Hi @Taejun-Kwon ,

Fault manager has dependency over app binary separation. PR #2747 
So if the app binary separation is not there, fault manager can not be enabled.

So I think this check is not needed, as number of apps can be anything not fixed to 2

Reply:
I have change it to initialize at the starting of fault manager.